### ISSUE TITLE
Treeview's multiselect keyboard controls should follow the Aria Practices Guidelines

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#6702]
The Treeview's multiselect keyboard functionality should follow the Aria Practices Guidelines [as detailed on this page](https://w3c.github.io/aria-practices/#keyboard-interaction-23).

[via @carmacleod](https://github.com/carbon-design-system/carbon/issues/5180#issuecomment-677671935)

> Note that the TreeView keyboard behaviors section in the Aria Practices Guide has a subsection for Selection in multi-select trees that is missing from this issue, but should be followed when the tree is multiselectable. You won't be able to implement the "Recommended selection model" because you want unmodified Space key to select and "activate" (i.e. perform the default action, just like the Enter key). So you need to look at implementing the "Alternative selection model".

**Comment by dakahn** [comment]
closing in favor or #6792
